### DISCUSSION TITLE
RefWidth and Height for Polygon

### DISCUSSION THREAD

**Discussion by CodeSippr** [clientIO/joint#1893] [Q&A]
Hi, 
I am desperate with the polygon element. I would like to add a second (inner) border. With elements like the rectangle, you can easily do this with refx/y and then refHeight/Width. For the Polygon, the Height and Width parameters don't work for me. Does anyone have an idea how I can still create an inner border?  

Thanks

**Comment by kumilingus** [comment]
Forget about `refX`, `refY`, `refWidth`, `refHeight`. They've been deprecated.

Start using [calc()](https://resources.jointjs.com/docs/jointjs/v3.6/joint.html#dia.attributes.intro) expression. See [examples](https://jsfiddle.net/kumilingus/97dwxqv3/).

**Comment by CodeSippr** [comment]
Thanks. I didn't realize that.

But when I use the calc function in d of a path element I get the following error. Do you have an idea how I can fix this?

<img width="416" alt="Bildschirm­foto 2022-11-15 um 15 12 47" src="https://user-images.githubusercontent.com/46285156/201941053-d3b6442a-9fd6-4bbb-8837-8fbf00c62828.png">

**Reply by kumilingus** [reply]
Upgrade to `v3.6` or use `calc(0.5 * w)` instead of `calc(w / 2)` (division was not supported prior `v3.6`).

**Comment by CodeSippr** [comment]
Now it works. Thank you. Is it also possible to set the x and y at path Elements without writing the calc Function in the path d? :)

**Reply by kumilingus** [reply]
You need to use the `transform` attribute with `SVGPathElement`.
```js
el.attr('body/transform', `translate(${x},${y})`); 
```
